transcom/mymove
# kickback on B-17958 identify/create calculation values 1 - 6

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/TeamRoom.mvc/Show/848239)

## Summary

In this PR the calculations for values 1-6 from the Feature Agility Ticket E-04525 are implemented.
Planned Move Start Date
Actual Move Date
Miles
Estimated Weight
Actual Net Weight
Pro Gear Weight (Customer)
Pro Gear Weight (Spouse)
Gross Incentive

### How to test
_Prereqs:
-run the server and client on the local dev environment
-create a move with a PPM shipment associated (not from test harness) and including Pro gear weight for customer and spouse
-open the Postman app_
-ensure Postman has cookies set up to match what is showing in your dev tools in the browser while logged in and looking at the PPM shipment on the MTO Shipment tab_

1. In the DB run the following SQL query (replacing the ```3VF8R9``` with the move locator ID associated with your move) to collect the ppm shipment ID.
```
select m.locator as move_code
       , m.id as move_id
       , ms.id as mto_shipment_id
       , ps.id as ppm_shipment_id
       , ms.shipment_type
       , o.gbloc
  from moves m
       inner join mto_shipments ms
               on m.id = ms.move_id
       inner join ppm_shipments ps
               on ms.id = ps.shipment_id
       inner join orders o
               on m.orders_id = o.id
where 1 = 1
   and m.locator = '3VF8R9'
   and ms.shipment_type = 'PPM'
order by m."locator"
         , ms.id
         , ms.shipment_type
```
_Note: It is important that the move/ppm is in the ppm closeout phase. The way I was able to get this done is that once everything has been approved and we are to the point the the customer has clicked the "download incentive packet" button and before clicking the "save and continue" button, you find the ppm shipment inside the ppm_shipments table with the following SQL query and manually change the status to WAITING_ON_CUSTOMER before proceeding with the customer process on the move_
```
select * from ppm_shipments ps where id = 'ba8b132e-b266-4698-a545-4f56505d3e68'
```
2. In the Postman app and on the path, make sure the ppm shipment ID you collected from the previous step is where it is supposed to go in the path.
```http://officelocal:3000{{baseUrl}}/ppm-shipments/:ppmShipmentId/closeout```
3. In the DB make sure the status of the PPM in the ppm_shipments table is set to NEEDS_PAYMENT_APPROVAL
4. Click the Send button in the Postman app.
5. Your result should be the whole ppmCloseout object with populated fields for the following (as long as those fields were entered, submitted, and approved):
-Planned Move Start Date
-Actual Move Date
-Miles
-Estimated Weight
-Actual Net Weight
-Pro Gear Weight (Customer)
-Pro Gear Weight (Spouse)
-Gross Incentive
